src/network/tcpsocket.c

```diff
@@ -422,6 +429,22 @@ void send_tcp(char *msg) {
     }
 }
 
+void check_tcp_timeout() {
+    static time_t last_check_time = 0;
+    struct client *cl, *tmp;
+    time_t now = time(0);
+    if (now - last_check_time >= 10 || now - last_check_time <= -10) {
```

Comment:
this `last_check_time ` is use for limiting the check rate (means that check every 10s)
but this now run in timer

the `update_tcp_connections` timer call every  `timeout_config.update_tcp_con` seconds
where timeout_config.update_tcp_con is 10 in default config

maybe need more think on this.